### PR TITLE
docs: Update CONTRIBUTING to reflect metadata usage

### PR DESCRIPTION
**Pull request by StephenWeatherford**
Update to reflect the fact that `metadata description` is now recommended, plus related guidance.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -59,18 +59,25 @@ The `metadata.json` file contains metadata of the module including `name`, `desc
...
+
 #### `main.bicep`
 
 The `main.bicep` file is the public interface of the module. When authoring `main.bicep`, make sure to provide a description for each parameter and output. You are free to create other Bicep files inside the module folder and reference them as local modules in `main.bicep` if needed. You may also reference other registry modules to help build your module. If you do so, make sure to add them as external references with specific version numbers. You should not reference other registry modules through local file path, since they may get updated overtime. The `main.bicep` most follow various static code analysis checks, such as including descriptions on every parameter and output. This will be used to automatically generate the `README.md`.
 
 ```bicep
+metadata name = 'Sample module'
+metadata description = '''Sample module description. This should be a short description of the functionality of the module.
+A more detailed description can be included in the README.md file.'''
```
Comment (shenglol) [inline CONTRIBUTING.md:78]: Just to make sure, we'll add metadata `details` to `main.bicep` once the `brm` tool is updated, right?

Comment (StephenWeatherford) [reply CONTRIBUTING.md:78]: No, the decision was the opposite. From Teams (luckily I wrote it down, too confusing to remember):
> Decision on "summary/description" vs "description/details".  We decided to go ahead and switch terminology from the current summary/description to "description/details" (as we had decided late last week).  This means the following:
>  
> 1) The "summary" property from metadata.json, which is a short documentation string will be switched to using "metadata description" in the bicep file.
> 
> 2) "Description" section in README.md will be renamed to "Details".

Work tracked here: https://github.com/Azure/bicep/issues/10810

I can't change Description in README.md until I update the tool.

**Review by shenglol — APPROVED** [review]

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -59,18 +59,25 @@ The `metadata.json` file contains metadata of the module including `name`, `desc
   // The name of the module (10 - 60 characters).
   "name": "Sample module",
   // The description of the module (10 - 1000 characters).
-  "description": "Sample module description",
+  "summary": "Sample module description. This should be a short description of the functionality of the module. A more detailed description can be included in the README.md file.",
```
Comment (shenglol) [inline CONTRIBUTING.md:62]: Could you also help update $schema to v2?

Comment (StephenWeatherford) [reply CONTRIBUTING.md:62]: done

**Pull request merged**